Fix health check behavior during cluster modification

bernardjkim opened the pull request:
### Description
This PR modifies a few health checks (`timedrift`, `nethealth`, `ping`). These healtch checks will no longer report a failed probe if the check does not run to completion.

- The `timedrift` check will no longer report a failed probe in situations described in https://github.com/gravitational/gravity/issues/1595.
- The `nethealth` check will no longer report a failed probe if the `kube-apiserver` is down. This scenario doesn't necessarily indicate a overlay network issue and instead just adds noise to the status.
- The `ping` check will no longer report a failed probe if coordinates are not found.

### Linked tickets and other PRs
* Updates https://github.com/gravitational/gravity/issues/1595, https://github.com/gravitational/gravity/issues/1552

### Testing done
**Verify modified health checks still report success probes**

**Verify `timedrift` is still detected.**
- Running a local cluster on VMs. We can create time drift by suspending a VM and resuming after some time.
- Should remote access go offline if timedrift is high?
```
[vagrant@node-1 ~]$ sudo gravity status
Cluster name:           dev.test
Cluster status:         degraded
Cluster image:          telekube, version 7.0.5
Gravity version:        7.0.5 (client) / 7.0.5 (server)
[...]
Cluster nodes:
    Masters:
        * node-2 / 172.28.128.102 / node
            Status:             degraded
            [×]                 time drift between 172.28.128.102 and 172.28.128.103 is higher than the allowed threshold of 300ms: -1m42.871109596s
            Remote access:      online
        * node-3 / 172.28.128.103 / node
            Status:             degraded
            [×]                 time drift between 172.28.128.103 and 172.28.128.101 is higher than the allowed threshold of 300ms: 1m42.90854089s
            [×]                 time drift between 172.28.128.103 and 172.28.128.102 is higher than the allowed threshold of 300ms: 1m42.86881878s
            Remote access:      offline
        * node-1 / 172.28.128.101 / node
            Status:             degraded
            [×]                 time drift between 172.28.128.101 and 172.28.128.103 is higher than the allowed threshold of 300ms: -1m42.908975939s
            Remote access:      online
[ERROR]: degraded
```

**Verify nethealth still detects blocked udp.**
```
[vagrant@node-1 ~]$ sudo iptables -A INPUT -p udp -s 172.28.128.102 -j DROP
[vagrant@node-1 ~]$ sudo gravity status
Cluster name:           dev.test
Cluster status:         active
Cluster image:          telekube, version 7.0.5
Gravity version:        7.0.5 (client) / 7.0.5 (server)
[...]
Cluster nodes:
    Masters:
        * node-2 / 172.28.128.102 / node
            Status:             healthy
            [!]                 overlay packet loss for node 172.28.128.101 is higher than the allowed threshold of 20% (current packet loss at 100%)
            Remote access:      online
        * node-3 / 172.28.128.103 / node
            Status:             healthy
            Remote access:      online
        * node-1 / 172.28.128.101 / node
            Status:             healthy
            [!]                 overlay packet loss for node 172.28.128.102 is higher than the allowed threshold of 20% (current packet loss at 100%)
            Remote access:      online
```

**Verify high ping latency is still detected**
- Use `tc` to simulate high latency (30ms) and verify ping check reports warning.
```
[vagrant@node-1 ~]$ sudo gravity status
Cluster name:           dev.test
Cluster status:         active
Cluster image:          telekube, version 7.0.5
Gravity version:        7.0.5 (client) / 7.0.5 (server)
[...]
Cluster nodes:
    Masters:
        * node-2 / 172.28.128.102 / node
            Status:             healthy
            [!]                 ping between 172_28_128_102.dev.test and 172_28_128_101.dev.test is higher than the allowed threshold of 15ms (ping latency at 33ms)
            Remote access:      online
        * node-3 / 172.28.128.103 / node
            Status:             healthy
            [!]                 ping between 172_28_128_103.dev.test and 172_28_128_101.dev.test is higher than the allowed threshold of 15ms (ping latency at 31ms)
            Remote access:      online
        * node-1 / 172.28.128.101 / node
            Status:             healthy
            [!]                 ping between 172_28_128_101.dev.test and 172_28_128_103.dev.test is higher than the allowed threshold of 15ms (ping latency at 31ms)
            [!]                 ping between 172_28_128_101.dev.test and 172_28_128_102.dev.test is higher than the allowed threshold of 15ms (ping latency at 33ms)
            Remote access:      online
```

bernardjkim committed:
Log timedrift/nethealth verification failure

r0mant reviewed (commented):
@bernardjkim Looks good - could you please also check if "ping" checker is affected by the same issue? I think one of those linked tickets also mentioned ping check.

bernardjkim committed:
Update ping check

- Ping check also no longer reports failed probe unless check runs to
completion.

r0mant reviewed (approved):
@bernardjkim Looks good, let's please make sure that these checkers do not regress before merging i.e. that they still catch issues they're supposed to :)

bernardjkim committed:
Modify probe reported by ping check

- ns -> ms
- Reported as warning

a-palchikov commented on monitoring/ping.go:
```diff
@@ -290,9 +290,10 @@ func (c *pingChecker) calculateRTT(serfClient agent.SerfClient, self, node serf.
-		Status: pb.Probe_Failed,
+		Detail: fmt.Sprintf("ping between %s and %s is higher than the allowed threshold of %dms",
+			c.self.Name, node, latencyThreshold.Milliseconds()),
+		Error:    fmt.Sprintf("ping latency at %dms", (latency / int64(time.Millisecond/time.Nanosecond))),
```
simpler: `latency/time.Millisecond`

bernardjkim committed:
Simply latency to ms conversion
